### PR TITLE
[CAMEL-18916] camel-xslt-saxon: allow logger injection

### PR DESCRIPTION
**Pull request by ammachado**
# Description

camel-xslt-saxon: allow logger injection

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

**Comment by davsclaus** [comment]
Can we use another name than messageConsumer - that smells like EIPish (camel message and consumer).

Its a logging listener or something.

**Review by davsclaus — APPROVED** [review]

File: components/camel-xslt/src/main/java/org/apache/camel/component/xslt/XsltBuilder.java [hunk 1]
```diff
@@ -463,7 +481,7 @@ protected void configureTransformer(Transformer transformer, Exchange exchange)
         addParameters(transformer, getParameters());
         transformer.setParameter("exchange", exchange);
         transformer.setParameter("in", exchange.getIn());
-        transformer.setParameter("out", exchange.getOut());
+        transformer.setParameter("out", exchange.getMessage());
```
Comment (davsclaus) [inline components/camel-xslt/src/main/java/org/apache/camel/component/xslt/XsltBuilder.java:484]: This is not related to this PR and it should be as before

**Review by davsclaus — CHANGES_REQUESTED** [review]

File: components/camel-xslt-saxon/src/main/java/org/apache/camel/component/xslt/saxon/XsltSaxonEndpoint.java [hunk 2]
```diff
@@ -209,8 +209,7 @@ protected XsltSaxonBuilder createXsltBuilder() throws Exception {
             }
         }
 
-        if (factory instanceof TransformerFactoryImpl) {
-            TransformerFactoryImpl tf = (TransformerFactoryImpl) factory;
+        if (factory instanceof TransformerFactoryImpl tf) {
```
Comment (davsclaus) [inline components/camel-xslt-saxon/src/main/java/org/apache/camel/component/xslt/saxon/XsltSaxonEndpoint.java:212]: This code style is only java 17 and as such this feature cannot be backported to 3.x. 

This PR is not related to this change, so its better to keep code as before, as it makes backporting bugs and other PRs easier from v4 to v3

Comment (ammachado) [reply components/camel-xslt-saxon/src/main/java/org/apache/camel/component/xslt/saxon/XsltSaxonEndpoint.java]: I'll revert this.

File: components/camel-xslt-saxon/src/main/java/org/apache/camel/component/xslt/saxon/XsltSaxonBuilder.java [hunk 3]
```diff
@@ -56,4 +69,66 @@ public void setAllowStAX(boolean allowStAX) {
     protected XmlSourceHandlerFactoryImpl createXmlSourceHandlerFactoryImpl() {
         return new SaxonXmlSourceHandlerFactoryImpl();
     }
+
+    @Override
+    protected Templates createTemplates(TransformerFactory factory, Source source) throws TransformerConfigurationException {
+        final Templates templates = super.createTemplates(factory, source);
+        if (templates instanceof TemplatesImpl templatesImpl && getXsltMessageLogger() != null) {
```
Comment (davsclaus) [inline components/camel-xslt-saxon/src/main/java/org/apache/camel/component/xslt/saxon/XsltSaxonBuilder.java:76]: Sorry again one of those java 17 only, if this feature should be backported to 3.x

Comment (ammachado) [reply components/camel-xslt-saxon/src/main/java/org/apache/camel/component/xslt/saxon/XsltSaxonBuilder.java]: Fixed.

**Review by davsclaus — APPROVED** [review]
